### PR TITLE
Added org.tensorflow.Tensor.create() methods that are an order of mag…

### PR DESCRIPTION
**Pull request by w-adamski**
…nitude faster on android.

### CODE REVIEW

**Comment by asimshankar** [comment]
@w-adamski : Thanks for the contribution! Did you see the note about the CLA? We can only look at the change after the CLA has been signed. Thanks!

**Comment by w-adamski** [comment]
Yes, I've seen the note about CLA, but apparently signing it by a corporation takes more time than signing by an individual.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @asimshankar, @protoget: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by asimshankar** [comment]
@w-adamski : Wasn't sure if your last comment meant that you do intend to figure out the CLA with your employer or if you're abandoning this PR. Could you clarify? Thanks!

**Comment by w-adamski** [comment]
As far as I know the CLA will be signed. Possibly (but it's not certain) this month.

**Comment by w-adamski** [comment]
I signed it!

**Comment by asimshankar** [comment]
@sjamesr : Mind taking a look? Seems reasonable to me

**Review by sjamesr — APPROVED** [review]

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

File: tensorflow/java/src/main/java/org/tensorflow/Tensor.java [hunk 1]
```diff
@@ -224,6 +224,74 @@
...
+   * On Android this method is preferred over {@link #create(long[], java.nio.IntBuffer) create}
+   * due to Android's ByteBuffer performance issues.
+   *
+   * @param shape the tensor shape.
+   * @param data an array containing the tensor data.
+   * @throws IllegalArgumentException If the tensor shape is not compatible with the array
+   */
+  public static Tensor<Integer> create(long[] shape, int[] data) {
+    Tensor<Integer> t = allocateForBuffer(DataType.INT32, shape, data.length);
+    setArray(t.nativeHandle, data);
```
Comment (rthadur) [inline tensorflow/java/src/main/java/org/tensorflow/Tensor.java]: error: cannot find symbol
    setArray(t.nativeHandle, data);
              ^
  symbol:   variable nativeHandle
  location: variable t of type Tensor<Integer>

File: tensorflow/java/src/main/java/org/tensorflow/Tensor.java [hunk 2]
```diff
@@ -224,6 +224,74 @@
...
+   * On Android this method is preferred over {@link #create(long[], java.nio.FloatBuffer) create}
+   * due to Android's ByteBuffer performance issues.
+   *
+   * @param shape the tensor shape.
+   * @param data an array containing the tensor data.
+   * @throws IllegalArgumentException If the tensor shape is not compatible with the array
+   */
+  public static Tensor<Float> create(long[] shape, float[] data) {
+    Tensor<Float> t = allocateForBuffer(DataType.FLOAT, shape, data.length);
+    setArray(t.nativeHandle, data);
```
Comment (rthadur) [inline tensorflow/java/src/main/java/org/tensorflow/Tensor.java]: error: cannot find symbol
    setArray(t.nativeHandle, data);
              ^
  symbol:   variable nativeHandle
  location: variable t of type Tensor<Integer>

File: tensorflow/java/src/main/java/org/tensorflow/Tensor.java [hunk 3]
```diff
@@ -224,6 +224,74 @@
...
+   * On Android this method is preferred over {@link #create(long[], java.nio.DoubleBuffer) create}
+   * due to Android's ByteBuffer performance issues.
+   *
+   * @param shape the tensor shape.
+   * @param data an array containing the tensor data.
+   * @throws IllegalArgumentException If the tensor shape is not compatible with the array
+   */
+  public static Tensor<Double> create(long[] shape, double[] data) {
+    Tensor<Double> t = allocateForBuffer(DataType.DOUBLE, shape, data.length);
+    setArray(t.nativeHandle, data);
```
Comment (rthadur) [inline tensorflow/java/src/main/java/org/tensorflow/Tensor.java]: same as above

File: tensorflow/java/src/main/java/org/tensorflow/Tensor.java [hunk 4]
```diff
@@ -224,6 +224,74 @@
...
+   * On Android this method is preferred over {@link #create(long[], java.nio.LongBuffer) create}
+   * due to Android's ByteBuffer performance issues.
+   *
+   * @param shape the tensor shape.
+   * @param data an array containing the tensor data.
+   * @throws IllegalArgumentException If the tensor shape is not compatible with the array
+   */
+  public static Tensor<Long> create(long[] shape, long[] data) {
+    Tensor<Long> t = allocateForBuffer(DataType.INT64, shape, data.length);
+    setArray(t.nativeHandle, data);
```
Comment (rthadur) [inline tensorflow/java/src/main/java/org/tensorflow/Tensor.java]: same as above

**Review by rthadur — CHANGES_REQUESTED** [review]
please address internal test failures , thank you

**Comment by herbakamil** [comment]
This PR has currently very low priority for us so if anyone needs that feel free to update it / use it in your own PR.

Anyway I will add it into the bottom of our backlog so maybe some day we'll go back to it if it makes sense.

**Comment by rthadur** [comment]
@herbakamil thank you for letting us know , please let us know once you are ready with new changes i can reopen it .

**Pull request closed without merging**